### ISSUE TITLE
Deparse doesn't like each $ref

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#11366]
Migrated from [rt.perl.org#91008](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=91008) (status was 'resolved')

Searchable as RT91008$

**Comment by p5pRT** [comment]
### From @nwc10

#### Created by @nwc10



$ ~/Sandpit/5140\-32/bin/perl5\.14\.0 \-MO=Deparse \-e 'while \(my $i = each $h\) \{\}'
unexpected OP\_REACH at /Users/nick/Sandpit/5140\-32/lib/perl5/5\.14\.0/B/Deparse\.pm line 1113\.
while \(my $i = XXX\) \{
&nbsp;    do \{
&nbsp;        \(\)
&nbsp;    \};
\}
\-e syntax OK

similarly

$ ~/Sandpit/5140\-32/bin/perl5\.14\.0 \-MO=Deparse \-e 'while \(my $i = keys $h\) \{\}'
unexpected OP\_RKEYS at /Users/nick/Sandpit/5140\-32/lib/perl5/5\.14\.0/B/Deparse\.pm line 1113\.
while \(my $i = XXX\) \{
&nbsp;    do \{
&nbsp;        \(\)
&nbsp;    \};
\}
\-e syntax OK
$ ~/Sandpit/5140\-32/bin/perl5\.14\.0 \-MO=Deparse \-e 'while \(my $i = values $h\) \{\}'
unexpected OP\_RVALUES at /Users/nick/Sandpit/5140\-32/lib/perl5/5\.14\.0/B/Deparse\.pm line 1113\.
while \(my $i = XXX\) \{
&nbsp;    do \{
&nbsp;        \(\)
&nbsp;    \};
\}
\-e syntax OK


\[sorry that I didn't think of trying this a couple of months ago\]

Nicholas Clark

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=low
    module=B::Deparse

Site configuration information for perl 5.14.0:

Configured by nick at Mon May 16 15:48:37 BST 2011.

Summary of my perl5 (revision 5 version 14 subversion 0) configuration:
  Derived from: a35ef416833511da752c4b5b836b7a8915712aab
  Platform:
    osname=darwin, osvers=10.7.0, archname=darwin-2level
    uname='darwin mouse-mill.local 10.7.0 darwin kernel version 10.7.0: sat jan 29 15:17:16 pst 2011; root:xnu-1504.9.37~1release_i386 i386 '
    config_args='-Dusedevel=y -Dcc=ccache gcc -Dld=gcc -Dcf_email=nick@ccl4.org -Dperladmin=nick@ccl4.org -Dinc_version_list=  -Dinc_version_list_init=0 -Doptimize=-Os -Uusethreads -Uuse64bitall -Uuselongdouble -Uusemymalloc -Duseperlio -Dprefix=~/Sandpit/5140-32 -Accflags=-m32 -Aldflags=-m32 -Alddlflags=-m32 -Dinstallman1dir=none -Dinstallman3dir=none -Dusevendorprefix -Dvendorprefix=~/Sandpit/vendor -Uuserelocatableinc -Uuseshrplib -de -Accflags=-DNO_MATHOMS -Umad'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='ccache gcc', ccflags ='-fno-common -DPERL_DARWIN -no-cpp-precomp -m32 -DNO_MATHOMS -fno-strict-aliasing -pipe -fstack-protector -I/opt/local/include',
    optimize='-Os',
    cppflags='-no-cpp-precomp -fno-common -DPERL_DARWIN -no-cpp-precomp -m32 -DNO_MATHOMS -fno-strict-aliasing -pipe -fstack-protector -I/opt/local/include'
    ccversion='', gccversion='4.2.1 (Apple Inc. build 5666) (dot 3)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='env MACOSX_DEPLOYMENT_TARGET=10.3 cc', ldflags =' -m32 -fstack-protector -L/usr/local/lib -L/opt/local/lib'
    libpth=/usr/local/lib /opt/local/lib /usr/lib
    libs=-lgdbm -ldbm -ldl -lm -lutil -lc
    perllibs=-ldl -lm -lutil -lc
    libc=/usr/lib/libc.dylib, so=dylib, useshrplib=false, libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=bundle, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags=' -bundle -undefined dynamic_lookup -m32 -L/usr/local/lib -L/opt/local/lib -fstack-protector'

Locally applied patches:
    


@INC for perl 5.14.0:
    /Users/nick/Sandpit/5140-32/lib/perl5/site_perl/5.14.0/darwin-2level
    /Users/nick/Sandpit/5140-32/lib/perl5/site_perl/5.14.0
    /Users/nick/Sandpit/vendor/lib/perl5/vendor_perl/5.14.0/darwin-2level
    /Users/nick/Sandpit/vendor/lib/perl5/vendor_perl/5.14.0
    /Users/nick/Sandpit/5140-32/lib/perl5/5.14.0/darwin-2level
    /Users/nick/Sandpit/5140-32/lib/perl5/5.14.0
    .


Environment for perl 5.14.0:
    DYLD_LIBRARY_PATH (unset)
    HOME=/Users/nick
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/opt/local/bin:/opt/local/sbin:/Users/nick/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/X11/bin:/usr/local/sbin:/sbin:/usr/sbin
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

Fixed by 09dcfa7

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'new' to 'resolved'
